### DISCUSSION TITLE
请教关于dns怎么填写

### DISCUSSION THREAD

**Discussion by Sereinfy** [kkkgo/PaoPaoGateWay#30] [问和答] [answered]
![23](https://github.com/kkkgo/PaoPaoGateWay/assets/122258839/f8d1522e-d51d-4572-9d9c-51fb62bf3c8d)
我现在这样配置的，觉得应该是合理的
1.泡泡dns的 dns怎么填写，假设我把客户机的dns改成网关ip

这样网关dns就应该改成 paopaodns的ip
我看到有人网关那里，填泡泡dns的ip
paopaodns的ip 又填的是网关。图2
![33](https://github.com/kkkgo/PaoPaoGateWay/assets/122258839/0c286e40-1d7f-40a6-b4e1-fa5f41ec51b1)
这样是能正常上网的，但是paopaodns的dns 合理吗，他们互相填了。还有人填127.0.0.1 ，这个会不会合理一点。
2.
然后fakeip网关那里，如果把客户机网关和dns都改成fakeip 的ip。就是全局
```
192.168.10.12
网关：192.168.10.8
dns：192.168.10.8
```
那么fakeip网关的dns 是192.168.10.6
泡泡dns的CUSTOM_FORWARD 是192.168.10.8：53
这种情况有没有必要把CUSTOM_FORWARD关了。
如果只改dns
```
192.168.10.12
网关：192.168.10.1
dns：192.168.10.8
```
![44](https://github.com/kkkgo/PaoPaoGateWay/assets/122258839/2e7cca02-abc2-4294-a969-e4a941327b40)
是这样。没搞明白咋工作的

解释起来应该挺复杂的，帮忙回复第一个问题就好

**Comment by kkkgo** [comment, marked as answer]
- 1、如果你想问PaoPaoDNS的宿主的DNS的话，填什么都不影响PaoPaoDNS正常工作，你填什么他都不会用你填的DNS，因为他能自己解析自己。
 - 2、你可以使用PaoPaoDNS作为其他DNS服务器的上游，然后使用其他DNS服务器，比如你所描述的网关使用PaoPaoDNS作为上游，客户端使用网关DNS，自然是不会有问题的。还可以套一个adguardhome之类的，都可以。
 - 3、fakeip网关填什么DNS并不会影响网关的工作，网卡属性的DNS只会用于7889端口获取配置，其他和DNS有关的配置都在ppgw.ini里面。
 - 4、如果你设备只需要全局代理，那么确实不需要CUSTOM_FORWARD，因为连PaoPaoDNS都没用上。如果你问的是会不会冲突，不会。

**Reply by kkkgo** [reply]
参考 https://github.com/kkkgo/PaoPaoDNS/discussions/24#discussioncomment-6112002
